hal/ia32/cpu.c

```diff
@@ -135,36 +135,16 @@ int hal_cpuReschedule(spinlock_t *spinlock, spinlock_ctx_t *scp)
 {
 	int err;
 
-	if (spinlock != NULL) {
-		hal_cpuGetCycles((void *)&spinlock->e);
-
-		/* Calculate maximum and minimum lock time */
-		if ((cycles_t)(spinlock->e - spinlock->b) > spinlock->dmax)
-			spinlock->dmax = spinlock->e - spinlock->b;
-
-		if (spinlock->e - spinlock->b < spinlock->dmin)
-			spinlock->dmin = spinlock->e - spinlock->b;
+	if (spinlock != NULL)
+	{
+		hal_spinlockClear(spinlock, scp);
 	}
 
 	__asm__ volatile (
-		"movl %1, %%eax;"
-		"cmp $0, %%eax;"
-		"je 1f;"
-
-		"movl %3, %%eax;"
-		"pushl %%eax;"
-		"xorl %%eax, %%eax;"
-		"incl %%eax;"
-		"xchgl %2, %%eax;"
-		"jmp 2f;"
-
-		"1:;"
 		"pushf;"
-
-		"2:;"
 		"pushl %%cs;"
 		"cli;"
-		"leal 3f, %%eax;"
+		"leal 1f, %%eax;"
 		"pushl %%eax;"
 		"movl $0, %%eax;"
```

Comment:
```suggestion
		"xorl %%eax, %%eax;"
```
Faster and less bytes.

Reply:
I agree, added in next commit.